### ISSUE TITLE
Changes to the Plugins API

### ISSUE THREAD

**Issue by Symbitic** [citation-js/citation-js#30]
I realize this is a breaking change, so it probably won't be added until you decide it's time for the 1.0 release.

This suggestion builds upon my pull request (#29). As ES6 modules become more and more widespread, global variables are increasingly frowned upon. Citation.js relies on globals for registering plugins: `require('@citation-js/plugin-bibtex')`

I would suggest changing the API to use explicit plugin registering. For example:
```js
import { Cite } from '@citation-js/core';
import bibtex from '@citation-js/plugin-bibtex';
import ris from '@citation-js/plugin-ris';

const cite = await Cite
  .use(bibtex)
  .use(ris)
  .async('Q21972834');
```

It would basically involve two steps:
1. Add a `.use()` method to `Cite` that calls `plugins.add` and then returns `this`.
2. Remove `plugins.add` from each plugin.

I would be willing to work on this, if you want. But first I have to make sure you even think it's a good idea.

**Comment by larsgw** [comment]
It shouldn't use globals, but rather a shared instance of `@citation-js/core`, which usually works (although it has its limits).

**Comment by Symbitic** [comment]
Shared instance is kind-of globals, at least here. When you import a plugin it registers itself with `@citation-js/core` instead of the user registering it.

It also has the side effect of meaning that Core must expose the plugin API for plugins to use. That leads to tighter coupling between the core Citation.js and the plugins. With this, you would no longer need to expose the plugin API from Core for plugins to use. Plugins would just need to export `ref`, `parsers`, and `format` (which they already do).

Nonetheless, it's completely your call. If you think what you have now is how you want it to be, that's good enough for me. Do you want me to close this issue?

**Comment by larsgw** [comment]
True. I'm probably going to leave this to `1.0` though, I want to finish up `0.4` at some point :slightly_smiling_face:.
